File: laravel/app/Services/Content/StreamService.php
<?php

namespace App\Services\Content;

use App\Models\Content\Stream;

class StreamService
{
    public function create(array $data): Stream
    {
        throw new \LogicException('Not implemented');
    }

    public function update(Stream $stream, array $data): Stream
    {
        throw new \LogicException('Not implemented');
    }
    public function goLive(int $id): void
    {
        $stream = Stream::findOrFail($id);
        $stream->goLive();
        $stream->save();
    }

    public function end(int $id): void
    {
        $stream = Stream::findOrFail($id);
        $stream->end();
        $stream->save();
    }

    public function updateViewerPeak(int $id, $count): void
    {
        $stream = Stream::findOrFail($id);
        $stream->updateViewerPeak($count);
        $stream->save();
    }
}

File: rails/app/services/content/stream_service.rb
module Content
  class StreamService
    def initialize(repository = Stream)
      @repository = repository
    end

    def create(attributes)
      raise NotImplementedError, "#{self.class}#create not implemented"
    end

    def update(entity, attributes)
      raise NotImplementedError, "#{self.class}#update not implemented"
    end

    def go_live(id)
      instance = Stream.find(id)
      instance.go_live()
      instance.save!
    end

    def end(id)
      instance = Stream.find(id)
      instance.end()
      instance.save!
    end

    def update_viewer_peak(id, count)
      instance = Stream.find(id)
      instance.update_viewer_peak(count)
      instance.save!
    end

    def duration_minutes(id)
      instance = Stream.find(id)
      result = instance.duration_minutes()
      instance.save!
      result
    end
  end
end
